File: src/test/java/com/widehouse/dnd5e/character/AbilityTest.java
package com.widehouse.dnd5e.character;

import static org.assertj.core.api.BDDAssertions.then;

import org.junit.jupiter.api.Test;

class AbilityTest {

    @Test
    void constructor_GivenStatScore_ThenCreateAbility() {
        Ability ability = new Ability(Stat.of(30), Stat.of(15), Stat.of(12), Stat.of(10), Stat.of(6), Stat.of(5));
        // then
        then(ability)
                .hasFieldOrPropertyWithValue("strength.score", 30)
                .hasFieldOrPropertyWithValue("dexterity.score", 15)
                .hasFieldOrPropertyWithValue("constitution.score", 12)
                .hasFieldOrPropertyWithValue("intelligence.score", 10)
                .hasFieldOrPropertyWithValue("wisdom.score", 6)
                .hasFieldOrPropertyWithValue("charisma.score", 5);
    }

    @Test
    void modifier_GivenStatScore_ThenReturnModifierValue() {
        Ability ability = new Ability(Stat.of(30), Stat.of(15), Stat.of(12), Stat.of(10), Stat.of(6), Stat.of(5));
        // then
        then(ability)
                .hasFieldOrPropertyWithValue("strength.modifier", 10)
                .hasFieldOrPropertyWithValue("dexterity.modifier", 2)
                .hasFieldOrPropertyWithValue("constitution.modifier", 1)
                .hasFieldOrPropertyWithValue("intelligence.modifier", 0)
                .hasFieldOrPropertyWithValue("wisdom.modifier", -2)
                .hasFieldOrPropertyWithValue("charisma.modifier", -3);
    }
}

File: src/test/kotlin/com/widehouse/dnd/character/AbilityTest.kt
package com.widehouse.dnd.character

import com.widehouse.dnd.character.ability.Dexterity
import com.widehouse.dnd.character.ability.Strength
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class AbilityTest : FunSpec({
    test("Ability modifier by score") {
        val str = Strength(15)
        str.score shouldBe 15
        str.modifier shouldBe 2

        val dex = Dexterity(13)
        dex.score shouldBe 13
        dex.modifier shouldBe 1
    }
})
